kafka/source/pkg/adapter/message.go

```diff
@@ -161,3 +176,79 @@ func getKeyTypeMapper(keyType string) func([]byte) interface{} {
 	}
 	return keyTypeMapper
 }
+
+func getAvroDecoder(schemaURL string) *AvroDecoder {
+	if schemaURL != "" {
+		return &AvroDecoder{
+			schemaURL: schemaURL,
+			codecMap:  make(map[int]*goavro.Codec),
+		}
+	}
+	return nil
+}
+
+func (decoder *AvroDecoder) getCodec(schemaID int) (*goavro.Codec, error) {
+	decoder.mutex.RLock()
+	if decoder.codecMap[schemaID] != nil {
+		return decoder.codecMap[schemaID], nil
+	}
+	decoder.mutex.RUnlock()
+
+	resp, err := nethttp.Get(fmt.Sprintf("%s/schemas/ids/%d", decoder.schemaURL, schemaID))
```

Comment:
yes Confluent Schema Registry

Reply:
@gunnarmorling any comments on this section ?

Reply:
Or perhaps @alesj might have some thoughts

Reply:
Hey, so I lack a bit of insight what this change is about, but ideally it would not be based on the assumption of any specific schema registry. E.g. the Apicurio registry also supports Avro and should be usable here. While it has a compatibility mode where its API should be the same as the one exposed by Confluent Schema Registry, Apicurio also has its own API. It'd be interesting to see what it'd take to support that one here.

Reply:
Look at this comment below https://github.com/knative/eventing-contrib/pull/1393#issuecomment-660847433